core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_contract.go

```diff
@@ -55,31 +51,70 @@ func (amc *AssetManagementContract) LockAsset(ctx contractapi.TransactionContext
         log.Error(err.Error())
         return false, err
     }
-    return amc.assetManagement.LockAsset(ctx.GetStub(), assetAgreement, lockInfo)
+    retVal, err := amc.assetManagement.LockAsset(ctx.GetStub(), assetAgreement, lockInfo)
+    if retVal && err == nil {
+        lockInfoHTLC := &common.AssetLockHTLC{}
+        err = proto.Unmarshal(lockInfo.LockInfo, lockInfoHTLC)
+        if err == nil {
+            contractInfo := &common.AssetContractHTLC{
+                Agreement: assetAgreement,
+                Lock: lockInfoHTLC,
+            }
+            contractInfoBytes, err := proto.Marshal(contractInfo)
+            if err == nil {
+                err = ctx.GetStub().SetEvent("LockAsset", contractInfoBytes)
+            }
+        }
+        if err != nil {
+            log.Warn("Unable to set 'LockAsset' event")
+            log.Warn(err.Error())
+        }
+    }
```

Comment:
note that the two reasons we are not guaranteeing events here are because 1) how we treat a failure of `setEvent` in the code above 2) we can only have a [single event in a transaction in Fabric](https://github.com/hyperledger/fabric/blob/1147977a810ac58c97bd9429b052cc9b56e496b0/docs/source/developapps/transactioncontext.md), so if the application chaincode that utilizes this (that we don't have control over), also emits an event, then ours would be replaced